File: backend/src/models/Lounge.js
const mongoose = require('mongoose');

const loungeSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Lounge name is required'],
    trim: true
  },
  ownerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: [true, 'Owner is required']
  },
  universityId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'University',
    required: [true, 'University is required']
  },
  campusId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Campus',
    required: [true, 'Campus is required']
  },
  description: {
    type: String,
    trim: true
  },
  logo: {
    type: String
  },
  bankAccount: {
    accountNumber: {
      type: String,
      required: [true, 'Bank account number is required']
    },
    bankName: {
      type: String,
      required: [true, 'Bank name is required']
    },
    accountHolderName: {
      type: String,
      required: [true, 'Account holder name is required']
    }
  },
  operatingHours: {
    opening: String,
    closing: String
  },
  isApproved: {
    type: Boolean,
    default: false
  },
  isActive: {
    type: Boolean,
    default: true
  },
  rating: {
    average: {
      type: Number,
      default: 0,
      min: 0,
      max: 5
    },
    count: {
      type: Number,
      default: 0
    }
  }
}, {
  timestamps: true
});

// Index for efficient queries
loungeSchema.index({ universityId: 1, campusId: 1, isActive: 1, isApproved: 1 });

module.exports = mongoose.model('Lounge', loungeSchema);

File: frontend/lib/models/lounge.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';

part 'lounge.g.dart';

@JsonSerializable()
class Lounge extends Equatable {
  final String id;
  final String name;
  final String ownerId;
  final String universityId;
  final String campusId;
  final String? description;
  final String? logo;
  final OperatingHours? operatingHours;
  final bool isApproved;
  final bool isActive;
  final Rating rating;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Lounge({
    required this.id,
    required this.name,
    required this.ownerId,
    required this.universityId,
    required this.campusId,
    this.description,
    this.logo,
    this.operatingHours,
    required this.isApproved,
    required this.isActive,
    required this.rating,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Lounge.fromJson(Map<String, dynamic> json) => _$LoungeFromJson(json);
  Map<String, dynamic> toJson() => _$LoungeToJson(this);

  @override
  List<Object?> get props => [
        id,
        name,
        ownerId,
        universityId,
        campusId,
        description,
        logo,
        operatingHours,
        isApproved,
        isActive,
        rating,
        createdAt,
        updatedAt,
      ];
}

@JsonSerializable()
class OperatingHours extends Equatable {
  final String? opening;
  final String? closing;

  const OperatingHours({
    this.opening,
    this.closing,
  });

  factory OperatingHours.fromJson(Map<String, dynamic> json) =>
      _$OperatingHoursFromJson(json);
  Map<String, dynamic> toJson() => _$OperatingHoursToJson(this);

  @override
  List<Object?> get props => [opening, closing];
}

@JsonSerializable()
class Rating extends Equatable {
  final double average;
  final int count;

  const Rating({
    required this.average,
    required this.count,
  });

  factory Rating.fromJson(Map<String, dynamic> json) => _$RatingFromJson(json);
  Map<String, dynamic> toJson() => _$RatingToJson(this);

  @override
  List<Object?> get props => [average, count];
}
